Allow syslog_json callback options to be set in an Ansible configurat…

paddynewman opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

The syslog_json documentation says that it supports options via an Ansible
configuration file. In fact, they can only be specified via environment
variables.

I've updated the module to use the standard "get_options" handling which means
that it can now support options via environment variables *or* the
configuration file.

Options can be set in the configuration file as follows:

```
callback_whitelist = syslog_json

[callback_syslog_json]
syslog_server = localhost
syslog_port = 514
syslog_facility = user
```

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
syslog_json callback plugin

paddynewman committed:
Allow syslog_json callback options to be set in an Ansible configuration file.

The syslog_json documentation says that it supports options via an Ansible
configuration file. In fact, they can only be specified via environment
variables.

I've updated the module to use the standard "get_options" handling which means
that it can now support options via environment variables *or* the
configuration file.

Options can be set in the configuration file as follows:

```
callback_whitelist = syslog_json

[callback_syslog_json]
syslog_server = localhost
syslog_port = 514
syslog_facility = user
```

bcoca commented on lib/ansible/plugins/callback/syslog_json.py:
```diff
@@ -14,25 +14,25 @@
+      - Before 2.9 only environment variables were available for configuration
     options:
-      server:
+      syslog_server:
```
you don't need to change the option names, if you do you'll need 'version_added' and keep the old ones with deprecation notice and as fallback

paddynewman replied:
Thanks. I'll set them back to their original values.

paddynewman committed:
Use the original, documented, names for the modules options.

In the documentation text change syslog_server to server, syslog_port to port
and syslog_facility to facility.

paddynewman committed:
Add an item to the changelog.

paddynewman committed:
Update 57232-syslog-json-configuration-options.yml

Fix a YAML syntax error / typo.
